components/movementsensor/adxl345/adxl345.go

```diff
@@ -41,14 +41,57 @@ var modelName = resource.NewDefaultModel("accel-adxl345")
 
 // AttrConfig is a description of how to find an ADXL345 accelerometer on the robot.
 type AttrConfig struct {
-	BoardName              string `json:"board"`
-	I2cBus                 string `json:"i2c_bus"`
-	UseAlternateI2CAddress bool   `json:"use_alternate_i2c_address,omitempty"`
+	BoardName              string                 `json:"board"`
+	I2cBus                 string                 `json:"i2c_bus"`
+	UseAlternateI2CAddress bool                   `json:"use_alternate_i2c_address,omitempty"`
+	InterruptPin1          InterruptPinAttrConfig `json:"interrupt_pin1,omitempty"`
+	InterruptPin2          InterruptPinAttrConfig `json:"interrupt_pin2,omitempty"`
+}
+
+// InterruptPinAttrConfig describes the interrupts associated with a given pin.
+type InterruptPinAttrConfig struct {
+	EchoInterrupt      string              `json:"echo_interrupt_pin"`
+	SingleTap          bool                `json:"single_tap,omitempty"`
+	FreeFall           bool                `json:"free_fall,omitempty"`
+	TapAttrConfig      *TapAttrConfig      `json:"tap_attributes,omitempty"`
+	FreeFallAttrConfig *FreeFallAttrConfig `json:"free_fall_attributes,omitempty"`
+}
+
+// TapAttrConfig is a description of the configs for tap registers.
+type TapAttrConfig struct {
+	ExcludeTapX bool `json:"exclude_tap_x,omitempty"`
```

Comment:
Are these named "Exclude" because false is the default for booleans? That would suck but I don't have a better suggestion.

Reply:
yes unfortunately